server/methods/core/groups.js

```diff
@@ -0,0 +1,207 @@
+import _ from "lodash";
+import { Meteor } from "meteor/meteor";
+import { check } from "meteor/check";
+import { Roles } from "meteor/alanning:roles";
+import { Logger } from "/server/api";
+import { Accounts, Shops } from "/lib/collections";
+
+/**
+ * Reaction Group Permission Methods
+ * TODO: Confirm if better to move this to shop.js since groups are on shops
+ */
+Meteor.methods({
+  /**
+   * group/createGroup
+   * @summary creates a new permission group for a shop
+   * creates permission group for a given shop with passed in roles
+   * @param {Object} groupData - info about group to create
+   * @param {String} groupData.name - name of the group to be created
+   * @param {Array} groupData.permissions - permissions to assign to the group being created
+   * @param {String} shopId - id of the shop the group belongs to
+   * @return {Object} - object.status of 200 on success or Error object on failure
+   */
+  "group/createGroup": function (groupData, shopId) {
+    check(groupData, Object);
+    check(groupData.name, String);
+    check(groupData.permissions, [String]);
+    check(shopId, String);
+
+    // must have needed permissions.. .TODO: Review who can create a group permission for a shop
+    if (!Roles.userIsInRole(Meteor.userId(), "admin", shopId)) {
+      throw new Meteor.Error(403, "Access Denied");
+    }
+
+    const shop = Shops.update({ _id: shopId }, { $addToSet: { groups: groupData } });
+
+    if (!shop) {
+      throw new Meteor.Error(400, "Bad request data");
```

Comment:
I _think_ (but might be wrong) that we've been using `(400, "Bad Request") for these types of errors.